owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/ShareActivity.kt

```diff
@@ -0,0 +1,385 @@
+import com.owncloud.android.presentation.ui.sharing.fragments.ShareFileFragment
+import com.owncloud.android.presentation.ui.sharing.fragments.ShareFragmentListener
+import com.owncloud.android.presentation.viewmodels.capabilities.OCCapabilityViewModel
+import com.owncloud.android.presentation.viewmodels.sharing.OCShareViewModel
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.ui.dialog.RemoveShareDialogFragment
+import com.owncloud.android.ui.utils.showDialogFragment
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+
+/**
+ * Activity for sharing files
+ */
+class ShareActivity : FileActivity(), ShareFragmentListener {
+    /**
+     * Shortcut to get access to the [ShareFileFragment] instance, if any
+     *
+     * @return A [ShareFileFragment] instance, or null
+     */
+    private val shareFileFragment: ShareFileFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_SHARE_FRAGMENT) as ShareFileFragment
+
+    /**
+     * Shortcut to get access to the [PublicShareDialogFragment] instance, if any
+     *
+     * @return A [PublicShareDialogFragment] instance, or null
+     */
+    private val publicShareFragment: PublicShareDialogFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_PUBLIC_SHARE_DIALOG_FRAGMENT) as PublicShareDialogFragment?
+
+    /**
+     * Shortcut to get access to the [EditPrivateShareFragment] instance, if any
+     *
+     * @return A [EditPrivateShareFragment] instance, or null
+     */
+    private val editPrivateShareFragment: EditPrivateShareFragment?
+        get() = supportFragmentManager.findFragmentByTag(TAG_EDIT_SHARE_FRAGMENT) as EditPrivateShareFragment?
+
+    // Needed for changing some UI elements in fragments before loading them, otherwise there will be UI glitches
+    private val ocCapabilityViewModel: OCCapabilityViewModel by viewModel {
+        parametersOf(
+            account
+        )
+    }
+
+    private val ocShareViewModel: OCShareViewModel by viewModel {
+        parametersOf(
+            file.remotePath,
+            account
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.share_activity)
+
+        // Set back button
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        val ft = supportFragmentManager.beginTransaction()
+
+        if (savedInstanceState == null && file != null && account != null) {
+            // Add Share fragment on first creation
+            val fragment = ShareFileFragment.newInstance(file, account!!)
+            ft.replace(
+                R.id.share_fragment_container, fragment,
+                TAG_SHARE_FRAGMENT
+            )
+            ft.commit()
+        }
+
+        observePrivateShareCreation()
+    }
+
+    public override fun onStop() {
+        super.onStop()
```

Comment:
you can remove this function, can't you?

Reply:
Sure